autopts/rtt.py

```diff
@@ -233,9 +243,23 @@ def start(self, buffer_name, log_filename, device_core, debugger_snr):
         self.log_file = open(log_filename, 'ab')
         self.rtt_reader.start(buffer_name, device_core, debugger_snr, self._on_line_read_callback, (self.log_file,))
 
-    def stop(self):
+    def stop_timer(self):
+        # timeout reached
+        self.rtt_reader.rtt_stop = True
+
+    def stop(self, reset, timeout=0.1):
         log("%s.%s", self.__class__, self.stop.__name__)
-        self.rtt_reader.stop()
+
+        if reset:
+            self.rtt_reader.stop()
+        else:
+            self.rtt_reader.test_end = True
+            # let's wait until stopping rtt reader; timeout is configurable
+            timer = threading.Timer(timeout, self.stop_timer)
+            timer.start()
+            while not self.rtt_reader.rtt_stop:
+                time.sleep(0.1)
```

Comment:
Even with `timeout=1` I still didn't all the logs when running locally. Had to do
```suggestion
            while not self.rtt_reader.rtt_stop:
                time.sleep(1)
```

As well

Reply:
So changing this helped to get all the logs?

Reply:
Yes, but I don't really understand why we would need a time.sleep here at all, if there's a timeout elsewhere.

Reply:
Please see updated version. made a few changes. Timer will be started when first empty packet is received and then cancelled if non-empty packet is received.